ms2query/query_from_sqlite_database.py

```diff
@@ -0,0 +1,189 @@
+"""
+Functions to obtain data from sqlite files.
+"""
+
+import sqlite3
+from typing import Dict, List
+from matchms.Spectrum import Spectrum
+import ast
+import pandas as pd
+import numpy as np
+import io
+
+
+def get_spectra_from_sqlite(sqlite_file_name: str,
+                            spectrum_id_list: List[str],
+                            table_name: str = "spectrum_data"
+                            ) -> List[Spectrum]:
+    """Returns a list with all spectra specified in spectrum_id_list
+
+    Args:
+    -------
+    sqlite_file_name:
+        File name of the sqlite file that contains the spectrum information
+    spectrum_id_list:
+        List of spectrum_id's of which the spectra objects should be returned
+    table_name:
+        Name of the table in the sqlite file that stores the spectrum data
+    """
+    # Converts TEXT to np.array when querying
+    sqlite3.register_converter("array", convert_array)
+
+    conn = sqlite3.connect(sqlite_file_name,
+                           detect_types=sqlite3.PARSE_DECLTYPES)
+
+    # Get all relevant data.
+    sqlite_command = f"""SELECT peaks, intensities, metadata FROM {table_name} 
+                    WHERE spectrum_id 
+                    IN ('{"', '".join(map(str, spectrum_id_list))}')"""
+    cur = conn.cursor()
+    cur.execute(sqlite_command)
+
+    # Convert to list of matchms.Spectrum
+    list_of_spectra = []
+    for result in cur:
+        peaks = result[0]
+        intensities = result[1]
+        metadata = ast.literal_eval(result[2])
+
+        list_of_spectra.append(Spectrum(mz=peaks,
+                                        intensities=intensities,
+                                        metadata=metadata))
+    conn.close()
+    return list_of_spectra
+
+
+def convert_array(text):
```

Comment:
`text` makes me think of a string. Maybe rename it to `text_array` (and add expected type) ?